lib/android/app/src/main/java/com/reactnativenavigation/views/component/ComponentLayout.java

```diff
@@ -50,13 +51,20 @@ public void start() {
         reactView.start();
     }
 
-	public void sendComponentStart() {
-		reactView.sendComponentStart(ComponentType.Component);
-	}
+    public void sendComponentWillStart() {
+        if (!willAppearSent)
```

Comment:
`willAppear` should be sent each time the component is going to be appeared as `didAppear` does